src/elements/content-sidebar/withSidebarAnnotations.js

```diff
@@ -69,7 +69,11 @@ export default function withSidebarAnnotations(
             const isTransitioningToAnnotationPath = activeAnnotationId && !isAnnotationsPath;
             const prevFileVersionId = getProp(prevMatch, 'params.fileVersionId');
 
-            if (annotation && prevAnnotation !== annotation) {
+            if (action === 'update_start' || action === 'update_end') {
+                this.updateAnnotation();
+            }
+
+            if ((action === 'create_start' || action === 'create_end') && annotation && prevAnnotation !== annotation) {
                 this.addAnnotation();
```

Comment:
Is it possible to combine add/update into a single method? It could detect whether the annotation already exists in cache and, if so, skip creating it anew? That would let us completely ignore `action` in this lifecycle step.

Reply:
If it was the only piece of logic to be added here I think your suggestion would make a lot of sense, but we will need to add support for 2 more actions in the coming days. One for deleting annotations and another for simply adding an annotation to the cache (and state) - when the annotation is added in `AnnotationThread` it needs to be "copied" to sidebar (annotation will not need to be created in sidebar as it's done in Preview integration). Having that in mind I believe that adding a clear distinction of which logic to follow based on action will be a lot more readable here.